cql3/statements/alter_table_statement.cc

```diff
@@ -142,7 +148,7 @@ static void validate_column_rename(data_dictionary::database db, const schema& s
     }
 }
 
-void alter_table_statement::add_column(const schema& schema, data_dictionary::table cf, schema_builder& cfm, std::vector<view_ptr>& view_updates, const column_identifier& column_name, const cql3_type validator, const column_definition* def, bool is_static) const {
+void alter_table_statement::add_column(const query_options&, const schema& schema, data_dictionary::table cf, schema_builder& cfm, std::vector<view_ptr>& view_updates, const column_identifier& column_name, const cql3_type validator, const column_definition* def, bool is_static) const {
```

Comment:
I finally understood why you needed these query_options (more on that below), but it seems that this particular function (and many of the functions below) doesn't use it, so why add this as a parameter?

Reply:
I've added it on all partial functions (not only on drop_column) so all of them have the same signature

Reply:
But what do you gain from this?

Reply:
See `alter_table_statement::prepare_schema_update`. We pass `add_column`, `alter_column`, `drop_column` methods as an argument to the `invoke_column_change_fn` lambda which expects the passed function to have a particular signature (see the `column_change_fn` typedef).

Reply:
Oh :-(
It's so ugly that we need to touch dozens of unrelated lines of code, but I guess there is no other way - if we want to allow that "USING TIMESTAMP ?" thing.

Reply:
There's no other way, and in any case it makes sense. It just happened that none of the schema altering statements allowed bind variables, so common infrastructure didn't allow for it.